ui/app/registration/controllers/patientCommonController.js

```diff
@@ -39,6 +45,49 @@ angular.module('bahmni.registration')
                 }
             });
 
+            $scope.checkDuplicatePatients = function () {
+                var patientGivenName = $scope.patient.givenName || '';
+                var patientLastName = $scope.patient.familyName || '';
+                var gender = $scope.patient.gender || '';
+                var birthDate = $scope.patient.birthdate || '';
+                var contactNumber = $scope.patient.PERSON_ATTRIBUTE_TYPE_PHONE_NUMBER || '';
+                if (birthDate != '') {
+                    birthDate = new Date(birthDate);
+                    birthDate = convertDate(birthDate);
+                }
+                var queryParams = patientGivenName + ' ' + patientLastName;
+
+                if (queryParams.length > 1) {
+                    patientService.searchDuplicatePatients(queryParams, gender, birthDate, contactNumber).then(function (response) {
+                        
+                        _.map(response.pageOfResults, function (result) {
+                            result.addressFieldValue = JSON.parse(result.addressFieldValue);
+                            result.extraIdentifiers = JSON.parse(result.extraIdentifiers);
+                            result.customAttribute = JSON.parse(result.customAttribute);
+                        });
+
+                        $rootScope.patients = response.pageOfResults;
+                        $rootScope.countNo = $rootScope.patients.length;
+                        $rootScope.visibleData = true;
+
+                        if ($rootScope.countNo == 0) {
+                            $rootScope.visibleData = false;
+                        }
+                    });
+                } else {
+                    $rootScope.visibleData = false;
+                }
+                
+            };
+
+            var convertDate = function (date) {
+                var yyyy = date.getFullYear().toString();
+                var mm = (date.getMonth() + 1).toString();
+                var dd = date.getDate().toString();
+                var mmChars = mm.split('');
+                var ddChars = dd.split('');
+                return yyyy + '-' + (mmChars[1] ? mm : "0" + mmChars[0]) + '-' + (ddChars[1] ? dd : "0" + ddChars[0]);
```

Comment:
@IipperMangesh The format is hardcoded here; could you please check if the format is set somewhere as a configuration? Could you please double check if there a better way to achieve this? Is there a reusable util function somewhere in bahmni (or from another library available in the project) that does the conversion?

Reply:
As per our need this should be there for checking present date and available date in our database.